scio-test/src/test/scala/com/spotify/scio/coders/instances/kryo/GrpcSerializersTest.scala

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2020 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio.coders.instances.kryo
+
+import com.spotify.scio.coders.{Coder, CoderMaterializer}
+import io.grpc.{Metadata, Status, StatusRuntimeException}
+import org.apache.beam.sdk.util.CoderUtils
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+import scala.collection.JavaConverters._
+import scala.util.{Failure, Try}
+
+class GrpcSerializersTest extends AnyFlatSpec with Matchers {
+
+  "StatusRuntimeException" should "roundtrip with all fields present" in {
+    val metadata = new Metadata()
+    metadata.put(Metadata.Key.of[String]("k", Metadata.ASCII_STRING_MARSHALLER), "v")
+
+    roundtrip(
+      Failure(
+        new StatusRuntimeException(
+          Status.OK.withCause(new RuntimeException("bar")).withDescription("bar"),
+          metadata
+        )
+      )
+    )
+  }
+
+  it should "roundtrip with all nullable fields absent" in {
+    roundtrip(Failure(new StatusRuntimeException(Status.OK)))
+  }
+
+  private def roundtrip(t: Try[String]): Unit = {
+    val kryoBCoder = CoderMaterializer.beamWithDefault(Coder[Try[String]])
```

Comment:
Something looks off here ... the usages above should have never compiled since they should be of type `Try[StatusRuntimeException]` 🤔

Reply:
I guess the only reason is somehow the type inference is off and it's something like `Failure[Nothing]`

Reply:
Ah no, it's supposed to be `Try[String]`, it's kind of a shortcut mimicking a real use case with AsyncLookupDoFn where the return type is something like `Try[GrpcResponse]`, which is either a `Success(GrpcResponse)` or `Failure(StatusRuntimeException)`. So I'm testing that (a) the Kryo serializer works and (b) that it's being picked up in a `Coder[Try[T]]` instead of the default Kryo coder registered for `Throwable`

Reply:
Ahh, right gotcha .. being `Try[String]` is confusing make it just `Try[T]` would be better.

Also, I think here we should be only testing `Coder[StatusRuntimeException]`. The semantic of picking a coder for a given `T` should be covered elsewhere.

Reply:
👍 

that's fair, I wrote the test like this because in most use cases we wouldn't have a static `Coder[StatusRuntimeException]` defined somewhere, the flow is like: AsyncLookupDoFn returns a `Try[GrpcResponse]` with a static `Coder[Try[GrpcResponse]]` --> result is determined to be a `Failure` so `FallbackCoder[Throwable]` is invoked --> Kryo deserializes the class name and correctly selects `StatusRuntimeExceptionSerializer` over `JavaSerializer` to deserialize the result. I can rework the test just to cover the last step :) (in a followup pr since this one is merged)

Reply:
Yup testing the last step is really what we need here 👍 

using `Coder[StatusRuntimeException]`, even if we don't foresee it's derivation, will just also make sure that we Fallback when we do.

However the thing that worries me more is the change on line L125. I would like to keep the default serializer otherwise we might have issues with subclasses.